v2/common/src/main/java/com/google/cloud/teleport/v2/cdc/mappers/DataStreamMapper.java

```diff
@@ -72,7 +72,7 @@ public TableId getTableId(TableRow input) {
         BigQueryConverters.formatStringTemplate(tableNameTemplate, input)
         .replaceAll("\\$", "_");
 
-    return TableId.of(datasetName, tableName);
+    return TableId.of(getProjectId(), datasetName, tableName);
```

Comment:
Unfortunately, this line seems to be causing some failures to internal tests:

```
java.lang.NullPointerException
com.google.common.base.Preconditions.checkNotNull(Preconditions.java:877)
com.google.cloud.bigquery.TableId.of(TableId.java:82)
com.google.cloud.teleport.v2.cdc.mappers.DataStreamMapper.getTableId(DataStreamMapper.java:75)
com.google.cloud.teleport.v2.cdc.mappers.DataStreamMapper.getTableId(DataStreamMapper.java:35)
com.google.cloud.teleport.v2.cdc.mappers.BigQueryMapper$1.apply(BigQueryMapper.java:214)
org.apache.beam.sdk.transforms.Contextful.lambda$fn$36334a93$1(Contextful.java:112)
org.apache.beam.sdk.transforms.MapElements$1.processElement(MapElements.java:143)
```

Since `outputProjectId` is not a required parameter, I think the factory method should be chosen based on whether or not `getProjectId()` returns null.

Reply:
Mmm.
This doesn't feel right. Does DataStreamMapper receive a null parameter in the constructor?
Also, the description of the _outputProject_ is as follows:

`The BigQuery Project ID. Default is the project for the Dataflow job.`

Reply:
Can we add the snippet below to _validateOptions_ method?

```
    String outputProjectId = options.getOutputProjectId();

    if (StringUtils.isBlank(outputProjectId)) {
      options.setOutputProjectId(options.as(GcpOptions.class).getProject());
    }
```

That way, I think the journey is more coherent, considering even the current description.

Reply:
Getting the value from GcpOptions does seem to be a better option, but I'd rather set the project id value in `DataStreamMapper` because:

1. Overwriting pipeline options in production code should be avoided.
2. We won't have to keep duplicating the same logic across templates.

That should be enough to fulfill the documented behavior of defaulting to the current GCP project.

Reply:
I agree.